src/vaadin-avatar-group.html

```diff
@@ -78,20 +115,140 @@
             items: {
               type: Array,
               observer: '__itemsChanged'
+            },
+
+            /**
+             * The maximum number of avatars to display.
+             * By default, all the avatars are displayed.
+             * When max is set, the overflowing avatars are grouped into one avatar.
+             */
+            max: {
+              type: Number
+            },
+
+            _maxReached: {
+              type: Boolean,
+              computed: '__computeMaxReached(items.length, max)'
+            },
+
+            _opened: {
+              type: Boolean,
+              observer: '__openedChanged',
+              value: false
             }
           };
         }
 
         ready() {
           super.ready();
 
-          this.addEventListener('iron-resize', this.__updateOffset);
+          this.__boundSetPosition = this.__setPosition.bind(this);
+
+          this.addEventListener('iron-resize', this._onResize.bind(this));
+
+          this._overlayElement = this.shadowRoot.querySelector('vaadin-avatar-group-overlay');
+        }
+
+        _onOverflowClick(e) {
+          e.stopPropagation();
+          if (this._opened) {
+            this.$.overlay.close();
+          } else if (!e.defaultPrevented) {
+            this._opened = true;
+          }
+        }
+
+        _onOverflowKeyDown(e) {
+          if (!this._opened) {
+            if (/^(Enter|SpaceBar|\s)$/.test(e.key)) {
+              e.preventDefault();
+              this._opened = true;
+            }
+          }
+        }
+
+        _onListKeyDown(event) {
+          if (event.key === 'Escape' || event.key === 'Esc') {
+            this._opened = false;
```

Comment:
I guess overlay should be closed whenever we are leaving tabs or we will get into the case where after tabbing out of the list-box you cannot properly close it anymore (until moving focus back):
![close-overlay](https://user-images.githubusercontent.com/29372419/86236996-a0cbde00-bba3-11ea-88ca-9470c8a223eb.gif)

Reply:
Good catch. Fixed, added missing tests.